crates/volta-core/src/hook/mod.rs

```diff
@@ -79,6 +153,48 @@ impl HookConfig {
             })?;
         src.parse()
     }
+
+    /// Creates a merged struct, with "right" having precedence over "left".
+    fn merge(left: Self, right: Self) -> Self {
+        Self {
+            node: if let Some(left_node) = left.node {
```

Comment:
I think these blocks can be done a little cleaner with a `match` instead of nested `if let` statements:

```rust
node: match (left.node, right.node) {
    (Some(left), Some(right)) => Some(ToolHooks::merge(left, right)),
    (Some(left), None) => Some(left),
    _ => right.node,
}
```

And similarly for the other blocks.

Reply:
Excellent, I had a feeling this wasn't fully idiomatic.

Reply:
I had to tweak this a bit to avoid a "use of moved value" issue, but it's still much better than before.